WebConfTech/website-2019
# Layouts

## What does this PR do?

The objective of this PR was to refactor the current page layout we were using for the landing and add a new one we could use for the new sections.

### 1: Constants everywhere

I move almost everything that shouldn't be hardcoded to constant files:

- `cfp`: The link to the Google form and the button title.
- `menu`: The list of routes we'll display on the menu.
- `seoData`: Yes, the data for SEO that'll be sent to the `SEO` component.
- `social`: The social network profiles' information.

Yes, `menu` shouldn't be a constant but a configuration, but since Gatsby automagically handles the routes, we don't have that. 

But don't worry, the `menu` will probably change on the future as we'll have to manually create routes for files, otherwise we'll have to name the files on spanish.

> @joelalejandro: Can you check on `social`, I copied the texts from the default page, but they're on english and I believe you said everything should be on spanish.

### 2: Social network links as a component

As the title says, that whole part of the default page has been moved to an specific component, and it uses the `social` constant to render the data.

### 3: SEO and the layout

I first move the implementation of the `SEO` component to the default layout and it now takes the data from the `seo` constant, but...I also added support for a `seoProps` (`Object`) prop, to overwrite the values from the constant.

One detail about the `seoProps` prop is that instead of overwriting the title, you can send a key `subtitle` and it will basically do this:

```js
`${SEO_DATA.title}${SEO_DATA.separator}${seoProps.subtitle}`
```

### 4: Footer component

I made a basic footer component with the CFP on the left (now using the data from the `cfp` constant) and the `SocialNetworkLinks` component on the right... basically the same visual we have right now.

### 5: Root styles on the default layout

The styles used to define the page size and background are now part of the default layout component, so any other page can have them by default.

### 6: Landing layout

Now the default layout is basically SEO + base styles; so I added a landing layout with that implements the default layout but adds the footer:

```
 _____________________________
|                             | 
|                             |
|           Content           |
|                             |
|_____________________________|
|                             |
|           Footer            |
|_____________________________|
```

### 8: Section layout

For the future inner sections of the site, I added this new layout that implements the navigation menu (next point):

```
 _____________________________
|         |                   | 
|         |                   | 
|   Menu  |     Content       |
|         |                   | 
|_________|___________________|
|                             |
|           Footer            |
|_____________________________|
```

> Right now, it has the the `Menu` component hardcoded there and the page needs to send the current route path in order to detect which option is selected; On the next iteration the menu will probably be on `children`, as a prop or a render prop, but looking to possible designs for the checkout process, we may need to make that part/slot configurable.

### 8: Menu component

I added a `Menu` component (which uses the `menu` constant) and defines most of the styles it uses from the design system.

**Not sure if it was the right approach**, but I made the mobile menu visibility toggle using `#` and `&:target`.

### 9: Random stuff

- I fixed some variables on the `AddEmailForm` that were using a hardcoded breakpoint.
- I added a `checkout.jsx` for @asermax to work on the next part.
- I added `temp` entry on the menu to reveal the checkout page.

### 10: Disclaimer

Yes, this probably should've been done over multiple PRs, [but...](https://www.youtube.com/watch?v=NPX48NpSRvo)

## How should it be tested manually?

1. Check that the landing page looks exactly as the live version.
2. Go to the `/checkout`, you should see the new layout with the menu on the left.
3. Resize to mobile and test the menu.
